athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/handlers/GlueMetadataHandler.java

```diff
@@ -275,9 +290,17 @@ protected ListTablesResponse doListTables(BlockAllocator blockAllocator, ListTab
         getTablesRequest.setDatabaseName(request.getSchemaName());
 
         Set<TableName> tables = new HashSet<>();
-        String nextToken = null;
+        String nextToken = request.getNextToken();
+        int pageSize = request.getPageSize();
         do {
             getTablesRequest.setNextToken(nextToken);
+            if (pageSize != UNLIMITED_PAGE_SIZE_VALUE) {
+                // Paginated requests will include the maxResults argument determined by the minimum value between the
+                // pageSize and the maximum results supported by Glue (as defined in the Glue API docs).
+                int maxResults = Math.min(pageSize, GET_TABLES_REQUEST_MAX_RESULTS);
+                getTablesRequest.setMaxResults(maxResults);
+                pageSize -= maxResults;
+            }
```

Comment:
I dont understand the -= being done here.  While not go with a simpler approach and keep calling Glue until you accomuilate pageSize or run out of results. This seems entirely unnecessary just to save a few rows from Glue.

Reply:
1. See comment above about Glue pagination.
2. This is the most straight-forward way to guarantees that no more than pageSize results are accumulated. I could have included a separate counter for the number of results retrieved, but that would require an additional variable that would would need to be subtracted from pageSize in order to determine the next max results size to retrieve in this iterative process. That seems a lot less intuitive than my solution.

Reply:
i don't follow. What example?